### PR TITLE
[PySpark] Add more functions such as ascii, asin, btrim, char, corr, ... and fix differences in ordering of null values between PySpark and DuckDB

### PR DESCRIPTION
**Pull request by binste**
For fixing the ordering of null values, I had to set `default_null_order` to `nulls_first` for the whole DuckDB connection. I've added the reasoning as comments in the code but let me know if I should expand on that. Thanks to the PySpark tests in CI, we at least now, that for the current test suite, this solves it.

### CODE REVIEW

**Review by Tishj — APPROVED** [review]
Thanks! 👍

File: tools/pythonpkg/duckdb/experimental/spark/context.py [hunk 1]
```diff
@@ -9,6 +9,11 @@
 class SparkContext:
     def __init__(self, master: str):
         self._connection = duckdb.connect(':memory:')
+        # We need this to align the ordering of ascending with Spark. For descending,
+        # Spark uses nulls last. Hence, in the API, we need to take care of that.
+        # Setting the ordering at least correctly for ascending is relevant
+        # for DataFrame.sort, see the comments there for more information.
+        self._connection.execute("set default_null_order='nulls first'")
```
Comment (Mytherin) [inline tools/pythonpkg/duckdb/experimental/spark/context.py:16]: Perhaps set this to `nulls_first_on_asc_last_on_desc` instead to fully match Sparks' behavior?

Comment (binste) [reply tools/pythonpkg/duckdb/experimental/spark/context.py:16]: Oh, that's exactly what I was looking for, thanks! I'll ping you once I've implemented it. That will simplify some things.

I'll see if I manage to open a PR to update the docs as well:
image

Comment (binste) [reply tools/pythonpkg/duckdb/experimental/spark/context.py]: Done (was able to revert most of the changes to `column.py` and `dataframe.py` thanks to this). 

Also opened a PR to update the docs https://github.com/duckdb/duckdb-web/pull/4034

PR is ready from my side.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**